mitodl/mitxpro
# 2447 Update e-commerce admin to support coupon discount types

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes https://github.com/mitodl/mitxpro/issues/2447

#### What's this PR do?
Updates coupon creation form to support different discount types like percent off and dollars off. Makes the relevant changes in the API payload and validations as well.

#### How should this be manually tested?

- visit `/ecommerce/admin/coupons/`
- Verify the discount type dropdown has options for percent-off and dollars-off.
- Verify the min and max values for the discount amount and relevant error messages.
- Verify that the correct amount and discount_type are sent in the payload. For percent-off amount is converted between 0.1-1.


#### Screenshots (if appropriate)
<img width="1000" alt="Screenshot 2022-11-09 at 3 00 04 PM" src="https://user-images.githubusercontent.com/52656433/200801243-59af5f9d-24fd-4514-b0f9-a1adc048a272.png">
<img width="1000" alt="Screenshot 2022-11-09 at 3 00 11 PM" src="https://user-images.githubusercontent.com/52656433/200801277-f769be0b-d93b-4a78-a30a-8343e704c193.png">
<img width="1000" alt="Screenshot 2022-11-09 at 3 00 22 PM" src="https://user-images.githubusercontent.com/52656433/200801285-a91e846d-7f9b-4cb2-84f0-ae4bf420944c.png">
